### PR TITLE
prov/sockets: Fix a bug where negative value is passed to strerror

### PR DESCRIPTION
**Pull request by gladkovdmitry17**
Signed-off-by: Dmitry Gladkov <dmitry.gladkov@intel.com>

### CODE REVIEW

**Comment by epaulson10** [comment]
Since you are returning the FI_E* versions of the error numbers, I think you should call fi_strerror instead of normal strerror in sock_ep_get_conn().

File: prov/sockets/src/sock_conn.c [hunk 1]
```diff
@@ -406,6 +406,8 @@ int sock_conn_listen(struct sock_ep_attr *ep_attr)
 	return -FI_EINVAL;
 }
 
+/* do NOT return negative value from this function, because
+ * its value is used as an argument to the strerror() function  */
```
Comment (shefty) [inline prov/sockets/src/sock_conn.c]: Almost all of the functions return 0 on success or negative value on error.  We should strive to be consistent with that.  I would rather change where the return value is passed to strerror().

Comment (gladkovdmitry17) [reply prov/sockets/src/sock_conn.c]: Yes, I agree. Applied your comment.

**Comment by gladkovdmitry17** [comment]
@epaulson10 Thanks for the comments! I agree with you.

This PR was updated as proposed. 
@shefty could you check this, please? Thanks!

**Pull request merged**